atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SpecificTableSweeper.java

```diff
@@ -128,11 +127,10 @@ void runOnceAndSaveResults(TableToSweep tableToSweep, SweepBatchConfig batchConf
 
     SweepResults runOneIteration(TableReference tableRef, byte[] startRow, SweepBatchConfig batchConfig) {
         try {
-            SweepResults results = sweepRunner.run(tableRef, batchConfig, startRow);
+            SweepResults results = sweepRunner.runWithMetricsUpdate(tableRef, batchConfig, startRow,
+                    this::updateMetricsDeleteBatch);
```

Comment:
I dislike passing anything to do with metrics down here.  We have a nice separation of sweep's actual work and it updating metrics which this breaks.  My preferred solution would be to include more fields in the SweepResults object (or a new object containing a SweepResults and other information) that hold this information and the metrics update code continues to live in one place.

Reply:
Failing that pass the `sweepMetricsManager` into the `sweepRunner` at construction time so the flow of code doesn't move back and forth between these 2 classes.